### ISSUE TITLE
Improve error messages for yaml parse errors

### ISSUE THREAD

**Issue by sjakobi** [commercialhaskell/stack#2374]
```
~/t/yaml-error $ cat stack.yaml 
resolver: lts-6.7
packages:
  - location: git: https://github.com/commercialhaskell/stack
~/t/yaml-error $ stack build
Could not parse '/home/simon/test/yaml-error/stack.yaml':
InvalidYaml (Just (YamlParseException {yamlProblem = "mapping values are not allowed in this context", yamlContext = "", yamlProblemMark = YamlMark {yamlIndex = 45, yamlLine = 2, yamlColumn = 17}}))
See http://docs.haskellstack.org/en/stable/yaml_configuration/.
```

At least we could use yaml's `prettyPrintParseException` to format the exception more nicely.

**Comment by WillSewell** [comment]
This can be closed right due to #2388 right?

**Comment by mgsloan** [comment]
I believe so, closing!
